tree/dataframe/inc/ROOT/RDF/RDatasetSpec.hxx

```diff
@@ -41,14 +42,20 @@ struct RDatasetSpec {
                 ULong64_t endEntry = 0, const std::vector<std::string> &defaultColumns = {},
                 const std::vector<std::string> &subTreenames = {})
       : fDatasetName(datasetName), fFileNameGlobs(std::vector<std::string>{fileName}), fStartEntry(startEntry),
-        fEndEntry(endEntry), fDefaultColumns(defaultColumns), fSubTreeNames(subTreenames)
+        fEndEntry((endEntry >= startEntry)
+                     ? endEntry
+                     : throw std::logic_error("RDatasetSpec: fEndEntry cannot be larger than fStartEntry.")),
```

Comment:
> RDatasetSpec: fEndEntry cannot be larger than fStartEntry.

On the contrary, the end must be larger than the start

Reply:
Also, the condition you wrote is with `>=`, we need to decide if the `=` part is valid or not. In principle it can since the `TTreeReader` won't care, but logically I feel like it shouldn't

Reply:
I have to fix the bug with specifying `Range(X,X)` anyway. Then we can decide if we want this support in `RDatasetSpec` or not

Reply:
https://github.com/root-project/root/issues/10540